caddyconfig/httpcaddyfile/builtins.go

```diff
@@ -181,6 +191,66 @@ func parseTLS(h Helper) ([]ConfigValue, error) {
 					cp.Curves = append(cp.Curves, h.Val())
 				}
 
+			case "clients":
+				ch := h.NewFromNextSegment() // new helper for client auth config
+				cp.ClientAuthentication = &caddytls.ClientAuthentication{}
+				for ch.Next() {
+					for ch.NextBlock(0) {
+						switch ch.Val() {
+						case "mode":
+							for ch.NextArg() {
+								cp.ClientAuthentication.Mode = ch.Val()
+							}
+						case "trusted_ca_certs":
+							for ch.NextArg() {
+								if _, err := caddytls.DecodeBase64DERCert(ch.Val()); err != nil {
+									return nil, h.Errf("cannot decode trusted CA certificate '%s'", ch.Val())
+								}
+								cp.ClientAuthentication.TrustedCACerts = append(cp.ClientAuthentication.TrustedCACerts, ch.Val())
+							}
+						case "trusted_ca_certs_file":
+							for ch.NextArg() {
+								filename := ch.Val()
+								// read PEM file
+								certDataPEM, err := ioutil.ReadFile(filename)
+								if err != nil {
+									return nil, err
+								}
+								// safe repack PEM file
+								block, _ := pem.Decode(certDataPEM)
+								if block == nil || block.Type != "CERTIFICATE" {
+									return nil, h.Errf("cannot decode trusted CA certificate file'%s'", ch.Val())
+								}
+								cp.ClientAuthentication.TrustedCACerts = append(cp.ClientAuthentication.TrustedCACerts, base64.StdEncoding.EncodeToString(block.Bytes))
```

Comment:
I think PEM blocks are just base64-encoded DER already, aren't they? Might not have to decode it; try just using block.Bytes directly and see if it works!